### ISSUE TITLE
Missing values for read by key with tablet moves

### ISSUE THREAD

**Issue by aphyr** [dgraph-io/dgraph#4540]
With tablet move operations involved, version 1.1.1 can occasionally return some sort of missing/blank/initial state for records--even those which have existed for quite some time. The effect appears transient: the record goes right back to its expected value in subsequent reads. @manishrjain says that there are known issues with tablet moves in 1.1.1, so this might be fixed already in master. I'm just writing this up so we have a record. :)

### What version of Dgraph are you using?

1.1.1

### Have you tried reproducing the issue with the latest release?

1.1.1 is the latest official release; I haven't seen this on 1.1.1-48-g157896305 yet.

### What is the hardware spec (RAM, OS)?

A 5-node EC2 m4.large cluster.

### Steps to reproduce the issue (command/config used to run Dgraph).

With Jepsen 3bff032adf3a4277e5cbbc2cd05ecec90c69f61e, run

```sh
lein run test --version 1.1.1 --concurrency 2n --nemesis move-tablet --time-limit 300 -w sequential
```

### Expected behaviour and actual result.

The Jepsen sequential workload increments a single record by reading it (by key), incrementing the associated value, then writing back that incremented value by UID. We expect that every process should (perhaps independently) observe a monotonically increasing value for these writes over time. However, this isn't always the case. In [this test run](https://github.com/dgraph-io/dgraph/files/4043382/20200109T194747.000Z.1.zip), two keys had a single instance of non-monotonic behavior. For instance, key 4:

```clj
{:sequential
 {:valid? false,
  :non-monotonic
  [[{:type :ok,
     :f :inc,
     :value 517,
     :process 29,
     :time 50839801681,
     :index 26356}
    {:type :ok,
     :f :read,
     :value 0,
     :process 29,
     :time 51492366936,
     :index 26666}]]},
```

Process 29 observed the value 517, and the immediately following value was 0. This doesn't mean that the value was *actually* zero; we [treat the absence of a record as 0](https://github.com/jepsen-io/jepsen/blob/3bff032adf3a4277e5cbbc2cd05ecec90c69f61e/dgraph/src/jepsen/dgraph/sequential.clj#L96-L98). I suspect that Dgraph failed to return any record, or perhaps returned a record with a nil value associated--I'll need to modify the test code to distinguish between these cases.

In this diagram, time flows down, and each process's operations fall in a single column. On the far right, we see process 29 increment 517, read 0, then read 524. No other process observed 0, nor did the counter reset. I think we might be looking at a problem similar to (possibly the same as?) #4534--the low account balances we saw in bank tests could potentially be explained by missing records as well.

![Screenshot from 2020-01-09 19-48-32](https://user-images.githubusercontent.com/3748/72116688-076cef00-3319-11ea-8fd5-537cb0a9c00d.png)

**Comment by aphyr** [comment]
This might the same issue as #4546, which encountered null values without any nemesis activity.

**Comment by danielmai** [comment]
```
Dgraph version   : v1.1.1-56-ge18986f1c
Dgraph SHA-256   : 36eecf05e86803b38a4a77da4a2410b11a7ec8d1c558d661a8024e4d0914d919
Commit SHA-1     : e18986f1c
Commit timestamp : 2020-01-11 17:49:16 -0800
Branch           : release/v1.1
Go version       : go1.13.5
```

This build passes the `sequential` test with `move-tablet` nemesis.

The [release/v1.1](https://github.com/dgraph-io/dgraph/tree/release/v1.1) branch (e18986f1c3feb0acfba15423827639f813eadbaf) contains the fix in #4496 and will be released as v1.1.2.

[20200112T194542.000Z.zip](https://github.com/dgraph-io/dgraph/files/4051444/20200112T194542.000Z.zip)

**Comment by aphyr** [comment]
@manishrjain traced #4546 to a different root cause, so that's a separate issue, but this might also be a dup of https://github.com/dgraph-io/dgraph/issues/4534.
